spec/java/util/StringBuilder.main.lsl

```diff
@@ -0,0 +1,1022 @@
+    {
+        _checkRangeSIOOBE(srcBegin, srcEnd, this.length);
+        var n: int = srcEnd - srcBegin;
+        val dstLength: int = action ARRAY_SIZE(dst);
+        _checkRange(dstBegin, dstBegin + n, dstLength);
+
+        var i: int = 0;
+        action LOOP_FOR(i, srcBegin, srcEnd, +1, _getChars_loop(i, dstBegin, dst));
+    }
+
+
+    @Phantom proc _getChars_loop(i: int, dstBegin: int, dst: array<char>): void
+    {
+        dst[dstBegin] = action CALL_METHOD(this.storage, "charAt", [i]);
+        dstBegin += 1;
+    }
+
+
+    // within java.lang.AbstractStringBuilder
+    fun *.codePointCount (@target self: StringBuilder, beginIndex: int, endIndex: int): int
+    {
+        if (beginIndex < 0 || endIndex > this.length || beginIndex > endIndex)
+            action THROW_NEW("java.lang.IndexOutOfBoundsException", []);
+
+        val count: int = endIndex - beginIndex;
+        val newStr: array<char> = action ARRAY_NEW("char", this.length);
+        var i: int = 0;
+        action LOOP_FOR(i, 0, this.length, +1, _newSubString_loop(i, newStr));
+
+        result = action DEBUG_DO("Character.codePointCount(newStr, beginIndex, count)");
+    }
+
+
+    // within java.lang.AbstractStringBuilder
+    fun *.codePointAt (@target self: StringBuilder, index: int): int
+    {
+        _checkIndex(index);
+
+        val newStr: array<char> = action ARRAY_NEW("char", this.length);
+        var i: int = 0;
+        action LOOP_FOR(i, 0, this.length, +1, _newSubString_loop(i, newStr));
+
+        result = action DEBUG_DO("Character.codePointAt(newStr, index, this.length)");
+    }
+
+
+    // within java.lang.AbstractStringBuilder
+    fun *.codePointBefore (@target self: StringBuilder, index: int): int
+    {
+        index -= 1;
+        _checkIndex(index);
+
+        val newStr: array<char> = action ARRAY_NEW("char", this.length);
+        var i: int = 0;
+        action LOOP_FOR(i, 0, this.length, +1, _newSubString_loop(i, newStr));
+
+        result = action DEBUG_DO("Character.codePointAt(newStr, index, this.length)");
+    }
+
+
+    // within java.lang.AbstractStringBuilder
+    fun *.offsetByCodePoints (@target self: StringBuilder, index: int, codePointOffset: int): int
+    {
+        _checkIndex(index);
+
+        result = action DEBUG_DO("Character.offsetByCodePoints(this.storage, index, codePointOffset)");
+    }
+
+
+    // within java.lang.AbstractStringBuilder
+    fun *.codePoints (@target self: StringBuilder): IntStream
+    {
+        // #todo: use custom stream implementation
+        result = action SYMBOLIC("java.util.stream.IntStream");
+        action ASSUME(result != null);
+    }
+
+
+    // within java.lang.AbstractStringBuilder
+    fun *.chars (@target self: StringBuilder): IntStream
+    {
+        // #todo: use custom stream implementation
+        result = action SYMBOLIC("java.util.stream.IntStream");
+        action ASSUME(result != null);
+    }
+
+
+    // within java.lang.Object
```

Comment:
The original class does not have any specific implementation for `hashCode`. Thus, the following can be removed.